modules/effects/src/effect_creator.ts

```diff
@@ -0,0 +1,59 @@
+import { Observable } from 'rxjs';
+import { Action } from '@ngrx/store';
+import { EffectMetadata } from './models';
+
+const CREATE_EFFECT_METADATA_KEY = '__@ngrx/effects_create__';
+
+export function createEffect<T extends Action>(
+  source: (() => Observable<T>),
```

Comment:
I had it first as a non action but it seemed a bit weird to me while changing the example-app,
because the non-dispatching effects were all `tap`s on the actions stream.

But now that you mention it, I do agree. It would make an effect harder to implement on a non-actions stream.

What do you think @brandonroberts ?

Reply:
I agree also. I thought the same, but I didn't mention it in the review 😞